pkg/receive/hashring.go

```diff
@@ -165,8 +165,10 @@ func newMultiHashring(cfg []HashringConfig) Hashring {
 // Hashrings are returned on the updates channel.
 // Which hashring to use for a tenant is determined
 // by the tenants field of the hashring configuration.
+// The updates chan is closed before exiting.
 func HashringFromConfig(ctx context.Context, updates chan<- Hashring, cw *ConfigWatcher) {
 	go cw.Run(ctx)
+	defer close(updates)
```

Comment:
isn't it a widely accepted that only  the sender should close the channel?

Reply:
agreed, I thought it would be confusing that we don't close it in the config-watcher (hashring) case but we do close it in the single node case, but I'll add a comment comment to explain.

Reply:
This goroutine IS the sender. In the single-node case, where there is no configuration watcher we have to close it ourselves.

Reply:
sorry for my ignorance, could you expand a bit more on that, 

I see no reason why it can't be closed somewhere in `runReceive` ?

Reply:
The reason we don't want to close it in `runReceive` is because by convention chans should always be closed by their sender to avoid races that `send on closed chan`. In this case, the `HashringFromConfig` func runs a goroutine that is the only sender of the given chan, and as such it should be the one to close it. Otherwise, we need to add another layer of synchronization to always ensure that the generator goroutine has returned *before* we close the given chan. This adds a lot of complexity for little benefit IMO.

Reply:
Ping @krasi-georgiev

Reply:
yes 100% correct, just got confused with the channel directions. Thanks for the patience!